sys/include/net/gnrc/ipv6/netif.h

```diff
@@ -264,6 +338,7 @@ void gnrc_ipv6_netif_remove(kernel_pid_t pid);
  */
 gnrc_ipv6_netif_t *gnrc_ipv6_netif_get(kernel_pid_t pid);
 
+#if (defined(MODULE_GNRC_NDP_ROUTER) || defined(MODULE_GNRC_SIXLOWPAN_ND_ROUTER))
```

Comment:
But to answer your question in a direct manner: To generate NOP macros in the else case.